archivesunleashed/auk-notebooks
# Reworks file outputs, resolves #44

## Rationale for Pull Request
#44 noted that some of our datathon attendees were finding the commenting out a bit difficult to follow – on review, this makes sense, as it's the only part of the notebook that requires the removal of a comment. I don't think outputting a text file is so catastrophic that we need to need a user to manually uncomment out a line, and hopefully they're at least reading the language around what they are executing.

And with #54 merged, we are all good to go on that front (thanks @greebie!).

## What this Changes
This PR makes two main changes:
- First, it removes the global variable for filenames because there are two different places we have outputs: years and domains. I think since they're writing files, and might be writing multiple files with different years, domains, etc., it doesn't make sense to use a variable and instead they should just provide the name of the file they're outputting to. Right now, you could easily imagine a user having to consistently hop between the executing cell and the set up cell.
- Second, it adds markdown explanations and new `nb.write_output` cells.

## How Should it be Tested
@greebie should make sure it also works on his end, and maybe we should quickly chat that you're ok with removing the global variable for file outputs. I think we should try to keep it as a simple as possible and this method removes the variable futzing.

## Example Screenshot
<img width="911" alt="Screen Shot 2019-04-25 at 8 17 28 AM" src="https://user-images.githubusercontent.com/3834704/56735114-93f76300-6732-11e9-95db-625cbdf06881.png">
